src/search/search_encoding.h

```diff
@@ -291,6 +359,92 @@ struct NumericFieldMetadata : IndexFieldMetadata {
   bool IsSortable() const override { return true; }
 };
 
+struct HnswVectorFieldMetadata : IndexFieldMetadata {
+  VectorType vector_type;
+  uint16_t dim;
+  DistanceMetric distance_metric;
+
+  uint32_t initial_cap = 500000;
+  uint16_t m = 16;
```

Comment:
It's not a good variable name.

Reply:
These metadata variable names are the same as [redis protocol](https://redis.io/docs/latest/develop/interact/search-and-query/advanced-concepts/vectors/). 

Also when browsing [related articles](https://www.pinecone.io/learn/series/faiss/hnsw/) regarding HNSW algorithm, it seems like it's a convention to use `m` or `m_L` referred to `max_num_outgoing_edges`. 

I feel like it might be good to be compatible with the protocol & convention, but still add comments beside them. Let me know if you have better ideas on how we can name it! <3